Add example solving the Stokes-Darcy problem

wenighttalk opened the pull request.

wenighttalk commented:
> Thanks @wenighttalk
> 
> Can you update the description of your PR with a brief description of the needed changes to support the Stokes-Darcy example?

Sure, I’ll add a quick summary to the PR description soon.

fdrmrc commented:
Thanks for the nice example. I'll review it soon!

fdrmrc commented on examples/PolyStokesDarcy.cc:
```diff
@@ -0,0 +1,2030 @@
+                                              bottom_left,
+                                              top_right);
+    domain_area = 1.0;
+    num_domain  = 16;
```
```suggestion
    num_cells  = 16;
```

fdrmrc reviewed (changes requested):
The indenter is complaining, but we can fix it later. In the meantime, I have only these minor suggestions.

wenighttalk commented:
> Thanks for addressing the comments. Ready to be merged once indentation issues are fixed (https://github.com/fdrmrc/Polydeal/actions/runs/16849374284/job/47733348448?pr=147).
> 
> Guotao, could you fix that? You can use the clang-format file available in the repo.

Sure, I’ll handle it later today.

fdrmrc commented:
@wenighttalk  I'll fix the indentation issues after merging this, no worries.

fdrmrc commented:
Indentation issues have been fixed in the main branch. Thanks @wenighttalk  for the nice example program! Looking forward to other nice test cases
